test/1.4/misc/linemarks.dml

```diff
@@ -0,0 +1,218 @@
+    }
+
+    // ... Unless it contains a declaration.
+    {
+        local int y;
+    }
+
+    if (true)
+    // multiple inner statements -- outermost mkCompound not collapsed
+    {
+        // inner Compounds are collapsed though, and made part of the outer
+        // NOT-LINEMARKED
+        {
+            assert true;
+            assert true;
+        // NOT-LINEMARKED
+        }
+        // ... unless they contain a declaration
+        {
+            assert true;
+            local int x;
+        }
+    }
+
+    if (true)
+    // Though in generated C you can observe braces, these stem from
+    // .toc_stmt() of the inner if statement. They do not come from the
+    // compound statement. That still gets collapsed.
+    // NOT-LINEMARKED
+    {
+        if (true)
+        // .toc_stmt() of empty braces emits ';', which gets linemarked to the
+        // leading brace
+        {
+            // NOT-LINEMARKED
+        }
+    // NOT-LINEMARKED
+    }
+    // else gets a dedicated linemark
+    else
+    ;
+    local int x = 3;
+    // Each inline param declaration has the site of its corresponding arg.
+    x = inline_m(
+        x
+        // NOT-LINEMARKED
+        );
+    assert x == 7;
+    // ... though constant folding can cause inline param declarations to
+    // disappear
+    x = inline_m(
+        // NOT-LINEMARKED
+        3);
+    assert x == 7;
+
+    // the "try" and "catch" keywords never get any linemarks
+    // NOT-LINEMARKED
+    try
+    {
+        local int y = 4;
+        throw;
+    }
+    // NOT-LINEMARKED
+    catch
+    // TryCatch uses .toc() of the catch block if the try block cannot
+    // fallthrough.
+    // In that case, the throw label uses the site of the catch statement,
+    // which means this brace DOES get a linemark, even though it's not
+    // emitted
+    {
+        assert true;
+        assert true;
+    // NOT-LINEMARKED
+    }
+
+    // NOT-LINEMARKED
+    try
+    // mkCompound collapse applies here.
+    // NOT-LINEMARKED
+    {
+        if (false)
+            throw;
+    // NOT-LINEMARKED
+    }
+    // NOT-LINEMARKED
+    catch
+    // TryCatch uses .toc_stmt() of the catch block if the try block could
+    // fallthrough.
+    {
+        assert true;
+        assert true;
+    }
+
+    x = 4;
+    // Linemark of do-while's "while" line is based on the condition expression
+    do
+        --x;
+    // NOT-LINEMARKED
+    while (
+        x != 0
+    // NOT-LINEMARKED
+    );
+}
+// END-LINEMARKS
+
+// Methods that are only inlined don't get linemarks for the braces
+inline method inline_m(inline x) -> (int) {
+    // BEGIN-LINEMARKS-FUNC _DML_M_init test_deinit
+    return x + 4;
+    // END-LINEMARKS
+}
+
+// destroy() is the only overridable method in DML 1.4 which is ad-hoc invoked
+// by DMLC via a codegen_inline_byname, and can possibly never be called
+// otherwise.
+method destroy() {
+    // BEGIN-LINEMARKS-FUNC test_deinit
+    local int x;
+    x = inline_m(x);
```

Comment:
we discussed elsewhere that
```
x = inline_m(
   x + 3 // this line should get a linemark
   );
```
... though it's arguably unimportant to test this, as we want to eventually retire inline methods.

Reply:
Already tested elsewhere, in `init()`.